theano/gof/tests/test_graph.py

```diff
@@ -311,3 +322,87 @@ def test_eval(self):
                 "variable must have cache after eval")
         self.assertFalse(hasattr(pickle.loads(pickle.dumps(self.w)), '_fn_cache'),
                 "temporary functions must not be serialized")
+
+
+################
+# autoname     #
+################
+class TestAutoName:
+
+    def test_auto_name(self):
+        ## Re-init counter
+        Variable.__count__ = count(0)
```

Comment:
It true that not resetting the value after the test would change a global value but I am not sure it's an issue. No test should rely on that global value being at some exact value 'x' at the beginning of the test because that value will change if you only run part of the tests or somehow change the order in which the tests are executed. I feel like both resetting the count and reading the current value and use it as a starting point are safe behaviors for a unit test. Could you give me a counter example of why it isn't?